File: app/src/main/java/iuliiaponomareva/eventum/data/NewsComparator.java
package iuliiaponomareva.eventum.data;

import java.io.Serializable;
import java.util.Comparator;

/**
 * Created by Julia on 15.04.2017.
 */
public class NewsComparator implements Comparator<News>, Serializable {

    @Override
    public int compare(News news1, News news2) {
        if ((news1.getPubDate() != null) && (news2.getPubDate() != null)) {
            return news2.getPubDate().compareTo(news1.getPubDate());
        } else {
            return Integer.MAX_VALUE;
        }
    }
}

File: app/src/main/java/iuliiaponomareva/eventum/data/NewsComparator.kt
package iuliiaponomareva.eventum.data

import java.io.Serializable
import java.util.*

/**
 * Created by Julia on 15.04.2017.
 */
class NewsComparator : Comparator<News>, Serializable {
    override fun compare(news1: News, news2: News): Int {
        return if (news1.pubDate != null && news2.pubDate != null) {
            news2.pubDate.compareTo(news1.pubDate)
        } else {
            Int.MAX_VALUE
        }
    }
}